src/store.js

```diff
@@ -1,61 +1,115 @@
-import NetMgr from './netMgr.js';
+import Vue from 'vue';
+import NetMgr from './services/netMgr.js';
 
 var store = {
-    user    : {"id" : 1},
-    trader  : {"id" : 1},
+    user: {"id": 1},
+    trader: {"id": 1},
     vouchers: [],
     recVouchers: [],
-    netMgr : NetMgr
+    netMgr: NetMgr,
+    auth: false
 };
 
-store.getRecVouchers = function() {
+/**
+ * Called from vue componenets, proxies logon process for them.
+ * @param userApiCreds
+ * @param success
+ * @param failure
+ */
+
+store.authenticate = function (userApiCreds, success, failure) {
+    this.netMgr.apiPost('/login', userApiCreds,
+        function (response) {
+            this.netMgr.setToken(response.data);
+            success();
+        }.bind(this),
+        function (error) {
+            if (failure) {
+                failure(error)
+            }
+        }
+    );
+};
+
+/**
+ * Logs the user off*
+ */
+store.unAuthenticate = function () {
+    this.netMgr.apiPost('/logout', null, function (response) {
+        this.netMgr.setToken(null);
+        this.auth = this.netMgr.isAuth();
+        // TODO: add store reset code.
+    }.bind(this));
+};
+
+
+/**
+ * Gets the server's idea of a trader's recorder voucher list
+ */
+store.getRecVouchers = function () {
 
     if (!navigator.onLine) {
         return false;
     }
 
-    this.netMgr.apiGet('traders/'+this.user.id +'/vouchers', function(response) {
-        var newVouchers = response.data;
-        newVouchers.sort(function(b,a) {
-            return new Date(a.updated_at).getTime() - new Date(b.updated_at).getTime()
-        });
-        newVouchers = response.data.map(function(v) {
-            return v.code;
-        });
-        // this is a callback. scoping of "this" gets broken, explicitly using "store".
-        store.mergeRecVouchers(newVouchers);
-    });
+    this.netMgr.apiGet('/traders/' + this.user.id + '/vouchers',
```

Comment:
It was the function for the old "recorded" list. It might well benefit from that now, will investigate.